django/db/backends/oracle/base.py

```diff
@@ -239,17 +241,69 @@ def __init__(self, *args, **kwargs):
         )
         self.features.can_return_columns_from_insert = use_returning_into
 
+    @property
+    def is_pool(self):
+        return self.settings_dict["OPTIONS"].get("pool")
+
+    @property
+    def pool(self):
+        if not self.is_pool:
+            return None
+
+        # Verify that we are not running with persistent connections
+        if self.settings_dict.get("CONN_MAX_AGE", 0) != 0:
+            raise ImproperlyConfigured(
+                "Pooling doesn't support persistent connections."
+            )
+
+        if self.alias not in self._connection_pools:
+            connect_kwargs = self.get_connection_params()
+            pool_options = connect_kwargs["pool"]
+
+            if pool_options is True:
+                connect_kwargs["max"] = 4
+            else:
+                for key, val in pool_options.items():
+                    connect_kwargs[key] = val
+                if not connect_kwargs.get("max"):
+                    connect_kwargs["max"] = 4
```

Comment:
Yes, the python-oracledb driver have max = 2 as default value. However, I've set it as 4 because the psycopg_pool has the default minimum and maximum values set as 4. Additionally, the test **test_pool_reuse** requires 3 connections, so setting the maximum to 2 would cause it to fail. Therefore, 4 seems to be a more suitable value, especially considering the psycopg_pool.

Reply:
Ah then I think `test_pool_reuse` should be setting the connection parameters similar to `tests.backends.postgresql.tests.Tests.test_connect_pool`. I don't think we need to have the max size defaults align to postgres and it means the code and docs is simpler

Reply:
I have removed the default value of "max" to be 4. Now, in case of default configuration, it will use whatever defaults are defined in oracledb.